anise-cli/src/main.rs

```diff
@@ -331,3 +253,58 @@ fn main() -> Result<(), CliErrors> {
         }
     }
 }
+
+fn truncate_daf_by_id<R>(
+    args::TruncateById {
+        input,
```

Comment:
I've not seen this syntax before. Does this unpack the struct on call? I read about this idea, but I didn't know that it was stabilized.

Reply:
Essentially yes, it's pattern I like for this sort of thing where the function consumes it's argument and the struct is defined primarily as a helper struct, all of whose fields you intend to use within the function's body. It's pretty common in the axum crate!